### PR TITLE
Feishu: close all connections before stopping HTTP servers

### LINKED ISSUE #48183: [Bug]: Feishu monitor state cleanup incomplete - potential memory leak in httpServers Map
## Summary

The Feishu plugin's monitor state cleanup in `monitor.state.ts` has a potential memory leak. When stopping the monitor, the httpServers are closed but the Map entries are deleted immediately without waiting for the server to fully close.

## Steps to Reproduce

1. Configure OpenClaw with multiple Feishu accounts
2. Start the gateway (Feishu monitor starts)
3. Repeatedly restart the gateway or reload Feishu accounts
4. Observe memory usage increasing over time

## Expected Behavior

When `stopFeishuMonitorState()` is called, all resources should be properly released:
- WebSocket clients should disconnect
- HTTP servers should close and release ports
- All Map entries should be cleared

## Actual Behavior

In `monitor.state.ts:145-150`:

The issue is that `server.close()` is asynchronous but the code doesn't wait for the 'close' event before clearing the Map. This can cause:
1. Memory leak - server object references retained
2. Port binding issues - port not fully released before next start
3. Resource leak - incomplete cleanup

## Environment

- OpenClaw version: 2026.3.x
- Feishu plugin: All versions

## Root Cause Analysis

In `monitor.state.ts`, the function `stopFeishuMonitorState()` calls `server.close()` synchronously but doesn't wait for the callback. This is a classic async cleanup bug.

## Proposed Solution

Use the 'close' event to ensure proper cleanup:

Or alternatively, use `server.closeAllConnections()` for faster cleanup in Node.js.

## Labels

bug, memory-leak, feishu

### PR DESCRIPTION
**Pull request by drvoss**
## Summary

- Problem: `stopFeishuMonitorState()` calls `server.close()` without terminating active connections first. On repeated account reloads, lingering connections prevent the server from fully shutting down, leaking resources and potentially blocking ports.
- Why it matters: Gateways with multiple Feishu accounts or frequent config reloads accumulate server objects that are not properly released.
- What changed: Added `server.closeAllConnections()` before `server.close()` in both the single-account and clear-all paths.
- What did NOT change (scope boundary): No changes to WS client lifecycle, webhook handling, or rate limiting. No signature changes.

> AI-assisted: This PR was authored with AI assistance (GitHub Copilot).

## Change Type (select all)

- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Docs
- [ ] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [x] Gateway / orchestration
- [ ] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [x] Integrations
- [ ] API / contracts
- [ ] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR

- Closes #48183

## User-visible / Behavior Changes

Feishu HTTP webhook servers now release all active connections on shutdown, preventing port-binding issues and resource leaks during account reloads.

## Security Impact (required)

- New permissions/capabilities? No
- Secrets/tokens handling changed? No
- New/changed network calls? No
- Command/tool execution surface changed? No
- Data access scope changed? No

### Environment

- OS: Any (Node.js 22+)
- Runtime/container: Node 22+
- Model/provider: N/A
- Integration/channel (if any): Feishu
- Relevant config (redacted): Multiple Feishu accounts configured

### Steps

1. Configure OpenClaw with multiple Feishu accounts
2. Start the gateway (Feishu monitor starts)
3. Repeatedly reload config or restart monitor
4. Observe that ports are not released until connections time out

### Expected

- All connections terminated and ports released immediately on monitor stop.

### Actual

- `server.close()` only stops accepting new connections; existing connections stay alive until they time out.

## Evidence

- [x] Failing test/log before + passing after

New tests verify `closeAllConnections()` is called before `close()` for both single-account and clear-all paths.

Test run:
```
pnpm test -- extensions/feishu/src/monitor.state.defaults.test.ts
4 passed (4)
```

## Human Verification (required)

- Verified scenarios: All 4 tests pass (2 existing + 2 new). Lint/format checks pass (`pnpm check`).
- Edge cases checked: Single-account cleanup and full clear-all cleanup both verified. Mock servers confirm both closeAllConnections and close are called.
- What you did **not** verify: Live multi-account Feishu gateway reload (no Feishu credentials available).

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

## Compatibility / Migration

- Backward compatible? Yes
- Config/env changes? No
- Migration needed? No

## Failure Recovery (if this breaks)

- How to disable/revert this change quickly: Revert the commit; servers will behave as before (close without force-terminating connections).
- Files/config to restore: `extensions/feishu/src/monitor.state.ts`
- Known bad symptoms reviewers should watch for: Unexpected connection drops during Feishu webhook processing (unlikely since this only fires on shutdown).

## Risks and Mitigations

- Risk: `closeAllConnections()` could terminate in-flight webhook requests during a config reload.
  - Mitigation: This is expected behavior during shutdown. The server is being stopped, so in-flight requests will not be processed anyway. The webhook sender (Feishu) will retry.

### CODE REVIEW

File: extensions/feishu/src/monitor.state.defaults.test.ts [hunk 1]
```diff
@@ -44,3 +47,38 @@ describe("feishu monitor state defaults", () => {
...
+  it("calls closeAllConnections before close for a single account", () => {
+    const server = mockServer();
+    httpServers.set("test-account", server);
+
+    stopFeishuMonitorState("test-account");
+
+    expect(server.closeAllConnections).toHaveBeenCalled();
+    expect(server.close).toHaveBeenCalled();
+    expect(httpServers.has("test-account")).toBe(false);
+  });
```
[the remark this answers is not in the document]
Comment (drvoss) [reply extensions/feishu/src/monitor.state.defaults.test.ts:64]: Fixed in 947697bfa: replaced separate `toHaveBeenCalled` assertions with a `callOrder` array that verifies exact invocation order via `expect(callOrder).toEqual(["closeAllConnections", "close"])`. Both single-account and clear-all tests now enforce ordering.
